python/ray/experimental/joblib/_raybackend.py

```diff
@@ -0,0 +1,52 @@
+import logging
+import ray
+from joblib._parallel_backends import MultiprocessingBackend
+from joblib.pool import MemmappingPool, PicklingPool
+from ray.experimental.multiprocessing.pool import Pool
+RAY_ADDRESS_ENV = "RAY_ADDRESS"
+logger = logging.getLogger(__name__)
+
+
+class RayBackend(MultiprocessingBackend):
+    """Ray backend uses ray, a system for scalable distributed computing.
+    more info are available here: https://ray.readthedocs.io
+    """
+
+    def configure(self,
+                  n_jobs=1,
+                  parallel=None,
+                  prefer=None,
+                  require=None,
+                  **memmappingpool_args):
+        # Make Ray Pool the father class of PicklingPool
+        PicklingPool.__bases__ = (Pool, )
+        MemmappingPool.__bases__ = (PicklingPool, )
+        """Use all available resources when n_jobs == -1. Must set RAY_ADDRESS
+        variable in the environment or run ray.init(address=..) to run on
+        multiple nodes.
+        """
+        if n_jobs == -1:
+            if not ray.is_initialized():
+                import os
+                if RAY_ADDRESS_ENV in os.environ:
+                    ray_address = os.environ[RAY_ADDRESS_ENV]
+                    logger.info(
+                        "Connecting to ray cluster at address='{}'".format(
+                            ray_address))
+                    ray.init(address=ray_address)
```

Comment:
why not set `address=auto`?

Reply:
what if someone wants to give the exact address? like `<ip_address>:<port>`

Reply:
They can run it in python - 

```
ray.init(address...)

# your code below
register_backend...
```

Reply:
I think it is more into preference. I prefer to enable users to also be able to set the address in their ENV. not sure why the current code does not support both ray.init(address...) from the outside and automatically doing it from the ENV?